Patch taken from upstream and edited for use in Fedora:

commit 6033275773313fe052f6e222321a8ec87587fbe6
Author: Julien Jorge <julien.jorge@iscool-e.com>
Date:   Thu Apr 24 20:39:18 2014 +0200

    Fix various errors pointed by clang on iOS.

diff --git a/claw/impl/box_2d.tpp b/claw/impl/box_2d.tpp
index e4b6936..71f3883 100644
--- a/claw/impl/box_2d.tpp
+++ b/claw/impl/box_2d.tpp
@@ -136,8 +136,8 @@ template<typename U>
 claw::math::box_2d<U> claw::math::box_2d<T>::cast_value_type_to() const
 {
   return claw::math::box_2d<U>
-    ( first_point.cast_value_type_to<U>(),
-      second_point.cast_value_type_to<U>() );
+    ( first_point.template cast_value_type_to<U>(),
+      second_point.template cast_value_type_to<U>() );
 } // box_2d::cast_value_type_to()
 
 /*----------------------------------------------------------------------------*/
diff --git a/claw/impl/rectangle.tpp b/claw/impl/rectangle.tpp
index 9e6f14a..3c3c8f3 100644
--- a/claw/impl/rectangle.tpp
+++ b/claw/impl/rectangle.tpp
@@ -140,7 +140,7 @@ template<typename U>
 claw::math::rectangle<U> claw::math::rectangle<T>::cast_value_type_to() const
 {
   return claw::math::rectangle<U>
-    ( position.cast_value_type_to<U>(), (U)width, (U)height );
+    ( position.template cast_value_type_to<U>(), (U)width, (U)height );
 } // rectangle::cast_value_type_to()
 
 /*----------------------------------------------------------------------------*/
